### PR TITLE
chore: add docblocks for type hinting and static analysis.

### PR DESCRIPTION
**Pull request by Carl-Tabuso**
### 🛠 Changes

- I added doc blocks for each model method.

### ✨ Context

- I initially documented each model method using two forward slashes doing something like this:

```php
    // returns the account of this employee
    public function account(): MorphOne
    {
        return $this->morphOne(User::class, 'account');
    }
```
- I opted to use a [DocBlock](https://docs.phpdoc.org/guide/getting-started/what-is-a-docblock.html#:~:text=A%20DocBlock%20is%20a%20piece,or%20other%20Structural%20Element%20is.) instead as a better way to do this. 

Using the same example above:

```php
    /**
     * Get the account associated with the guest.
     * 
     * @return \Illuminate\Database\Eloquent\Relations\MorphOne
     */
    public function account(): MorphOne
    {
        return $this->morphOne(User::class, 'account');
    }
```
- The comments improve the type-hinting of classes and methods/functions whenever we reference them somewhere else in the codebase.
- I decided to use the VSCode extension: **PHP DocBlocker** which automatically generates the documentation block for our classes and methods.
- I find this extension helpful and if you're interested, you may look it up on the vscode extensions.

### 🧠 Rationale

- It can help editors like VSCode perform static analysis to our code and indicate warnings/errors. 

### 🎥 Demos (Optional)

https://github.com/user-attachments/assets/1109368b-f4c0-4cfd-8097-70741daa3e85

- Check the demo above that shows the `account()` method of a morphable model being used within the `GoogleCallback` trait and how docblocking the method's **summary**, **description**, and **tags and annotations @**
 
### ✔️ Checklist

- [x] Is `staging` the base branch of this PR?
- [x] Do your changes not reveal sensitive information, such as secrets, API keys, etc?
- [x] Are there no erroneous console logs, debuggers, or leftover code in your changes?

### CODE REVIEW

**Review by C-Ivan-Bandilla — APPROVED** [review]
[is this the intentional?](https://github.com/POWERLANE-TEAM/emplo-hrms/pull/58/commits/e1b79f727833b5910edc1968629abbb1a6af61d0#:~:text=*%20%40return%20string-,%24coolstuff,-*/)

**Comment by Carl-Tabuso** [comment]
> https://github.com/POWERLANE-TEAM/emplo-hrms/commit/e1b79f727833b5910edc1968629abbb1a6af61d0#:~:text=*%20%40return%20string-,%24coolstuff,-*/

what's this?

File: app/Enums/AccountType.php [hunk 1]
```diff
@@ -8,7 +8,11 @@ enum AccountType: string
     case APPLICANT = 'applicant';
     case EMPLOYEE = 'employee';
 
-    // displays user-friendly account types in blade templates
+    /**
+     * Return user-friendly account type labels.
+     * 
+     * @return string $coolstuff
```
Comment (C-Ivan-Bandilla) [inline app/Enums/AccountType.php:14]: this one

Comment (Carl-Tabuso) [reply app/Enums/AccountType.php:14]: ahh. I was perhaps supposed to use that for demo.

**Comment by Carl-Tabuso** [comment]
can i merge this now?

**Pull request merged**